test/e2e/framework/pod/wait.go

```diff
@@ -168,6 +171,45 @@ func errorBadPodsStates(badPods []v1.Pod, desiredPods int, ns, desiredState stri
 	return TimeoutError(errStr)
 }
 
+// BeRunningNoRetries verifies that a pod starts running. It's a permanent
+// failure when the pod enters some other permanent phase.
+func BeRunningNoRetries() types.GomegaMatcher {
+	return gomega.And(
+		// This additional matcher checks for the final error condition.
+		gcustom.MakeMatcher(func(pod *v1.Pod) (bool, error) {
+			switch pod.Status.Phase {
+			case v1.PodFailed, v1.PodSucceeded:
+				return false, gomega.StopTrying(fmt.Sprintf("Expected pod to reach phase %q, got final phase %q instead.", v1.PodRunning, pod.Status.Phase))
+			default:
+				return true, nil
+			}
+		}),
+		BeRunning(),
```

Comment:
In this case here, or in general?

I defined it to achieve feature parity with existing functions ("WaitFor..PodRunning"). It might save an import in some (rare) cases. Overall I agree that it doesn't add much value and could drop it.

Reply:
yeah I mean generally

Reply:
Let's remove it. The explosion of `WaitFor*` functions with different names and subtle semantic differences has been a problem, we should strive to avoid that - thanks for reminding me! :grin: